src/store/item.ts

```diff
@@ -51,10 +53,20 @@ export const useItemStore = defineStore("item", {
 
     async saveData(podcastname: string, docname: string) {
       const docRef = doc(collection(db, podcastname), docname)
-      return setDoc(docRef, {
-        items: this.itemList,
-        slotTitles: this.slotTitleList,
-      })
+      try {
+        await updateDoc(docRef, {
+          items: this.itemList,
```

Comment:
The real point of this exercise is to use arrayUnion etc so you're only pushing the change not the whole doc each time

Reply:
I have tried arrayUnion but it duplicates the record.
would you please check the attached video?

https://user-images.githubusercontent.com/13918292/171849055-2efdd3a5-81ad-48ae-8045-8e6b7fbe2d43.mp4

Reply:
I have used arrayUnion/arrayRemove in creating/deleting an item 

Also, I have tried using arrayUnion in the update/drag item function but is not working well so I have not used arrayUnion in the update/drag item.